projects/angular/src/popover/common/popover-content.ts

```diff
@@ -5,175 +5,409 @@
-   * it is a low priority issue for now.
-   */
-  private addContent() {
-    // Create the view container
-    this.view = this.container.createEmbeddedView(this.template);
-    const [rootNode] = this.view.rootNodes;
-    this.smartEventsService.contentRef = rootNode; // So we know where/what to set close focus on
-    this.renderer.addClass(rootNode, 'clr-popover-content');
-    // Reset to the begining of the document to be available for sizing/positioning calculations.
-    // If we add new content to the bottom it triggers changes in the layout that may lead to false anchor
-    // coordinates values.
-    this.renderer.setStyle(rootNode, 'top', '0px');
-    this.renderer.setStyle(rootNode, 'left', '0px');
-    // We need to hide it during the calculation phase, while it's not yet finally positioned.
-    this.renderer.setStyle(rootNode, 'opacity', '0');
-    this.removeClickListenerFn = this.renderer.listen(rootNode, 'click', event => {
-      this.smartOpenService.openEvent = event;
-    });
-    this.view.rootNodes.forEach(node => {
-      this.renderer.appendChild(this.document.body, node);
-    });
-    // Mark for realingment on the next content-check cycle.
-    this.shouldRealign = true;
+  private getPositionStrategy() {
+    this.setPreferredPosition(); //Preferred position defined by consumer
+
+    // fetch all Scrolling Containers registered with CDK
+    let scrollableAncestors: CdkScrollable[];
+    if (this.popoverService.anchorElementRef) {
+      scrollableAncestors = this.scrollDispatcher.getAncestorScrollContainers(this.popoverService.anchorElementRef);
+    }
+
+    return this.overlay
+      .position()
+      .flexibleConnectedTo(this.popoverService.anchorElementRef)
+      .setOrigin(this.popoverService.anchorElementRef)
+      .withPush(true)
+      .withPositions([this.preferredPosition, ...this.popoverService.availablePositions])
+      .withFlexibleDimensions(true)
+      .withScrollableContainers(scrollableAncestors);
   }
 
-  private removeContent(): void {
-    if (!this.view) {
+  private closePopover() {
+    if (!this.popoverService.overlayRef) {
       return;
     }
-    if (this.removeClickListenerFn) {
-      this.removeClickListenerFn();
-      this.removeClickListenerFn = null;
+
+    this.removeOverlay();
+    this.popoverService.open = false;
+
+    const shouldFocusTrigger =
+      this.popoverService.popoverType !== ClrPopoverType.TOOLTIP &&
+      (document.activeElement === document.body ||
+        document.activeElement === this.popoverService.anchorElementRef?.nativeElement);
+
+    if (shouldFocusTrigger) {
+      this.popoverService.focusAnchor();
+    }
+  }
+
+  private showOverlay() {
+    // Get Scrollable Parents
+    this.scrollableParents = [
+      ...this.getScrollParents(this.popoverService.anchorElementRef?.nativeElement),
+      ...this.getScrollParents(this.popoverService.anchorElementRef?.nativeElement, 'x'),
+    ];
```

Comment:
If we go with the suggestion I gave.
```suggestion
    this.scrollableParents = this.getScrollParents(this.popoverService.anchorElementRef.nativeElement);
```

Reply:
Corrected in commit 7d9a087